src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,426 @@
+
+        const parsedURL = urlUtils.parseURL(inputURL);
+        if (!parsedURL) {
+            return undefined;
+        }
+        const server = this.getAllServers().find((server) => {
+            return equalUrlsIgnoringSubpath(parsedURL, server.url, ignoreScheme) && parsedURL.pathname.match(new RegExp(`^${server.url.pathname}(.+)?(/(.+))?$`));
+        });
+        if (!server) {
+            return undefined;
+        }
+        const tabs = this.getOrderedTabsForServer(server.id);
+
+        let selectedTab = tabs.find((tab) => tab && tab.name === TAB_MESSAGING);
+        tabs.
+            filter((tab) => tab && tab.name !== TAB_MESSAGING).
+            forEach((tab) => {
+                if (parsedURL.pathname.match(new RegExp(`^${tab.url.pathname}(/(.+))?`))) {
+                    selectedTab = tab;
+                }
+            });
+        return selectedTab;
+    }
+
+    updateServerOrder = (serverOrder: string[]) => {
+        log.debug('updateServerOrder', serverOrder);
+
+        this.serverOrder = serverOrder;
+        this.persistServers();
+    }
+
+    updateTabOrder = (serverId: string, tabOrder: string[]) => {
+        log.debug('ServerManager.updateTabOrder', serverId, tabOrder);
+
+        this.tabOrder.set(serverId, tabOrder);
+        this.persistServers();
+    }
+
+    addServer = (server: Team) => {
+        const newServer = new MattermostServer(server, false);
+        this.servers.set(newServer.id, newServer);
+        this.serverOrder.push(newServer.id);
+        const tabOrder: string[] = [];
+        getDefaultTabs().forEach((tab) => {
+            const newTab = this.getTabView(newServer, tab.name, tab.isOpen);
+            this.tabs.set(newTab.id, newTab);
+            tabOrder.push(newTab.id);
+        });
+        this.tabOrder.set(newServer.id, tabOrder);
+
+        this.emit(SERVERS_URL_MODIFIED, [newServer.id]);
+        this.persistServers();
+        return newServer;
+    }
+
+    editServer = (serverId: string, server: Team) => {
+        const existingServer = this.servers.get(serverId);
+        if (!existingServer) {
+            return;
+        }
+
+        let urlModified;
+        if (existingServer.url.toString() !== urlUtils.parseURL(server.url)?.toString()) {
+            urlModified = () => this.emit(SERVERS_URL_MODIFIED, [serverId]);
+        }
+        existingServer.name = server.name;
+        existingServer.updateURL(server.url);
+        this.servers.set(serverId, existingServer);
+
+        this.tabOrder.get(serverId)?.forEach((tabId) => {
+            const tab = this.tabs.get(tabId);
+            if (tab) {
+                tab.server = existingServer;
+                this.tabs.set(tabId, tab);
+            }
+        });
+
+        urlModified?.();
+        this.persistServers();
+    }
+
+    removeServer = (serverId: string) => {
+        this.tabOrder.get(serverId)?.forEach((tabId) => this.tabs.delete(tabId));
```

Comment:
I think this line here is irrelevant, no need for it as we are deleting the key from the map in the next line

Reply:
We're deleting from a different map in this line (that's what the `forEach` is for).
I suppose we could leave them in there but then they're never explicitly dereferenced and they might get stuck in memory.